File: Code Generation/responses/cpp/deepseek/easy/phone-number-prefix.cpp
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    bool phonePrefix(vector<string> numbers) {
        // Sort the numbers lexicographically
        sort(numbers.begin(), numbers.end());
        
        // Check each adjacent pair
        for (int i = 0; i < numbers.size() - 1; i++) {
            // If the next number starts with the current number, current is a prefix
            if (numbers[i + 1].find(numbers[i]) == 0) {
                return false;
            }
        }
        
        return true;
    }
};

File: Code Generation/responses/python/deepseek/easy/phone-number-prefix.py
from typing import List

class Solution:
    def phonePrefix(self, numbers: List[str]) -> bool:
        # Sort the numbers lexicographically
        numbers.sort()
        
        # Check each adjacent pair for prefix relationship
        for i in range(len(numbers) - 1):
            if numbers[i + 1].startswith(numbers[i]):
                return False
        
        return True